From 9eff25b3ea8401d51e26e97bb079ba6ecada3b0d Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:03:54 -0400
Subject: [fs] btrfs: fix double free in find_lock_delalloc_range

Message-id: <1407179109-13714-348-git-send-email-sandeen@redhat.com>
Patchwork-id: 87327
O-Subject: [RHEL7.1 PATCH 347/422] Btrfs: fix double free in find_lock_delalloc_range
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Chris Mason <clm@fb.com>

commit 7d78874273463a784759916fc3e0b4e2eb141c70
Author: Chris Mason <clm@fb.com>
Date:   Wed May 21 05:49:54 2014 -0700

    Btrfs: fix double free in find_lock_delalloc_range

    We need to NULL the cached_state after freeing it, otherwise
    we might free it again if find_delalloc_range doesn't find anything.

    Signed-off-by: Chris Mason <clm@fb.com>
    cc: stable@vger.kernel.org
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/extent_io.c b/fs/btrfs/extent_io.c
index d6689d3..7bcf6a8 100644
--- a/fs/btrfs/extent_io.c
+++ b/fs/btrfs/extent_io.c
@@ -1693,6 +1693,7 @@ again:
    * shortening the size of the delalloc range we're searching
    */
   free_extent_state(cached_state);
+  cached_state = NULL;
   if (!loops) {
    max_bytes = PAGE_CACHE_SIZE;
    loops = 1;
-- 
1.7.1